samples/crypto/mbedtls_benchmark/src/benchmark.c

```diff
@@ -179,7 +179,7 @@ do {                                                                  \
 	}                                                             \
 								      \
 	delta = k_cycle_get_32() - tsc;                               \
-	delta = SYS_CLOCK_HW_CYCLES_TO_NS(delta);                     \
```

Comment:
If you really want to avoid overflow, I guess `k_cycle_get_32` should be changed into `k_cycle_get_64`.

Reply:
I can't find this API, is it present in Zephyr headers?

Reply:
Oh right, I mixed that with `k_uptime_get` which has both a 32 and a 64-bit version. So looking at your change it seems the overflow doesn't happen at the cycle counter level, but in `SYS_CLOCK_HW_CYCLES_TO_NS`. Is it correct? If yes then your patch looks correct.

Reply:
Correct, it's the nanosecond conversion that caused the overflow. All in all, with nanosecond resolution, `u32_t` overflows after ~4 seconds.